### ISSUE TITLE
[Switch] Trying to go back to the RetroArch menu after ROM boot crashes RetroArch

### ISSUE THREAD

**Issue by ShadowOne333** [libretro/melonDS#54]
Downloaded yesterday's (August 11th) nightly of melonDS, and fired it up on my 8.1.0 Switch running Atmosphere with title redirection.

I do have the correct firmware.bin, bios7.bin and bios9.bin files in the system directory.
Managed to boot quite a few games just fine (although slow, but that's expected).

However, when trying to use the button combo to go back to the Quick Menu of RetroArch, the console crashes and throws out an Atmosphere (CFW) error.
This seems to be consistent with any ROM I try to boot.

**Comment by Datamats** [comment]
Unable to reproduce on the latest nightly, it might have been caused by some refactoring in libretro/RetroArch since it's statically compiled you get the newest version of RetroArch too

**Comment by ShadowOne333** [comment]
I got the August 13th nightly (since it seems the Aug 14th nightly is borked), and while I didn't get the same crash (as in an Atmosphere crash/system crash forcing reboot), it still crashes the application and forces the current app to be closed down.

For what it's worth, I'm using L3+R3 to access Quick Menu, with a PowerA wireless controller since my JoyCons are on shipping for drifting repairs.
What can I do to hopefully get rid of any possible bugged config for melonDS without deleting my entire RA configs?

**Comment by ShadowOne333** [comment]
Okay I managed to find the culprit of the issue.

MelonDS goes back to the RetroArch menu without problems when usinh yhe switch_thread Audio driver.

However, since I switched to the newly developed Audren driver this week, I forgot I had it on.
So if you switch the audio driver to switch_audren_thread, melonDS crashes everytime when trying to go back to the menu.

I tested this by only making this change, and got it consistently every time.
So the issue seems to be related to the Audren audio driver.
